addons/web/static/tests/views/record_tests.js

```diff
@@ -230,4 +232,170 @@ QUnit.module("Record Component", (hooks) => {
         assert.verifySteps(["record changed"]);
         assert.strictEqual(target.querySelector("[name='foo'] input").value, "753");
     });
+
+    QUnit.test("handles many2one fields", async function (assert) {
+        patchWithCleanup(AutoComplete, {
+            timeout: 0,
+        });
+
+        serverData.models = {
+            bar: {
+                records: [
+                    { id: 1, display_name: "bar1" },
+                    { id: 3, display_name: "abc" },
+                ],
+            },
+        };
+
+        class Parent extends Component {
+            setup() {
+                this.fields = {
+                    foo: {
+                        name: "foo",
+                        type: "many2one",
+                        relation: "bar",
+                    },
+                };
+                this.values = {
+                    foo: [1, undefined],
```

Comment:
not sure about this

Reply:
me neither. In fact I think several semantics of what values are are to be supported one way or another (at least in the long term)
For now, values can only be the equivalent of what the first read returns, hence the tuple for many2ones, If we are to support some other format for values, we'd need to implement some stuff in the RelationalModel that could handle already processed data or partially processed data.

Reply:
What bothers me here is that it's not really the format of what read returns. Read returns a pair `[id, display_name]`. Here you basically give an id in an array `[id]`. I'm not sure to know why it works :D

Reply:
from relational_model.js:
```
    async _loadMany2OneData(fieldName, value) {
        const relation = this.fields[fieldName].relation;
        const activeField = this.activeFields[fieldName];
        if (
            activeField &&
            !this.isInvisible(fieldName) &&
            value &&
            (!value[1] || activeField.options.always_reload)
        ) {
            const context = this.getFieldContext(fieldName);
            const result = await this.model.orm.nameGet(relation, [value[0]], { context });
            return result[0];
        }
        return value;
    }
 ```

if value && not value[1] => nameget

Reply:
ok we can move forward with this, but at some point (in RelationalModel v2?), we could support manyone values that are just an id, not an id inside an array. It would make more sense imo